src/test/g8/test

```diff
@@ -0,0 +1,21 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# make sure the jar is built
+> package
+$ exists target/dfdl-format-0.1.0-SNAPSHOT.jar
+
+# make sure the tests pass
+> test
```

Comment:
If this file is supposed to be a template, I didn't see any file like it in the generated project.  Is the sbt test command in `.github/workflows/main.yml` supposed to call this file?

Reply:
This is a [scripted](https://www.scala-sbt.org/1.x/docs/Testing-sbt-plugins.html) test file. These are usually used for testing plugins, but it can also be used for testing g8. Essentially what happens is g8 will run "sbt new" on itself, then use this file to run sbt commands and commands that test that those sbt commands work. So this essentially does "sbt new "..., then "sbt package" and makes sure it builds the jar file, then "sbt test" and makes sure those tests pass.

Reply:
Please add comment akin to what you just said, into the file, or if you prefer, into the README or somewhere.

Reply:
Added a comment in the README, I figure more people are going to see that there.

Also, I learned that this only allows  a single sanity check test. Kindof dissappointing, but it's a limitation of the giter8 plugin we use.